airflow/jobs/base_job.py

```diff
@@ -66,23 +76,23 @@ class BaseJob(Base, LoggingMixin):
         'polymorphic_identity': 'BaseJob'
     }
 
-    __table_args__ = (
-        Index('job_type_heart', job_type, latest_heartbeat),
-        Index('idx_job_state_heartbeat', state, latest_heartbeat),
-    )
-
     heartrate = conf.getfloat('scheduler', 'JOB_HEARTBEAT_SEC')
+    if conf.getboolean('heartbeat', 'redis_enabled'):
+        redis = Redis.from_url(conf.get('heartbeat', 'redis_url'))
+    else:
+        redis = None
 
     def __init__(
             self,
             executor=None,
             heartrate=None,
-            *args, **kwargs):
+            *args,
+            **kwargs):
         self.hostname = get_hostname()
         self.executor = executor or ExecutorLoader.get_default_executor()
         self.executor_class = executor.__class__.__name__
         self.start_date = timezone.utcnow()
-        self.latest_heartbeat = timezone.utcnow()
+        self._legacy_heartbeat = timezone.utcnow()
```

Comment:
Let's not call it "legacy" just because a user hasn't enabled it. It's not legacy, and not going to be removed.